### ISSUE TITLE
Clarification on Table 3 ("Inference-only") and input embedding procedure during generation

### ISSUE THREAD

**Issue by user50lab** [XiaoxinHe/G-Retriever#47]
Hi, thank you for sharing this excellent work!
I have two questions regarding the inference setup and reported results:
(1) About Table 3 (“Inference-only”)
When I run the command: 
`python inference.py --dataset expla_graphs --model_name inference_llm --llm_model_name 7b_chat --max_txt_len 0 --seed 0`
I obtained Test Accuracy = 0.6227, which is higher than the paper’s reported 0.5650.
However, when I run:
`python inference.py --dataset expla_graphs --model_name inference_llm --llm_model_name 7b_chat --seed 0`
(with --max_txt_len=512), the accuracy drops to 0.3267.
So, I would like to confirm:

> In Table 3 of the paper, does the “Inference-only” setting correspond to the Question-Only results (i.e., without textual description input)?

(2) About input embeddings during inference
In the inference() function, the code performs:
```
input_ids = descriptions.input_ids[i][:self.max_txt_len] + questions.input_ids[i] + eos_user_tokens.input_ids
inputs_embeds = self.word_embedding(torch.tensor(input_ids).to(self.model.device))
outputs = self.model.generate(inputs_embeds=inputs_embeds, ...)
```
rather than directly passing input_ids to generate().

> I would like to understand the reason behind using precomputed embeddings (inputs_embeds) here.
> Is this mainly to manually concatenate BOS/EOS tokens or special task-specific embeddings,
> or does it also serve another purpose?

Additionally,

> If a model’s generate() function does not support inputs_embeds as input,
> would it be valid to instead provide the corresponding input_ids to achieve equivalent behavior?
